dashboard/src/components/AppView/AppControls/AppControls.test.tsx

```diff
@@ -2,17 +2,46 @@ import { mount, shallow } from "enzyme";
 import context from "jest-plugin-context";
 import * as React from "react";
 import Modal from "react-modal";
-import { Redirect } from "react-router";
+import { Provider } from "react-redux";
+import configureMockStore, { MockStore } from "redux-mock-store";
+import thunk from "redux-thunk";
+
+import { IRelease } from "shared/types";
+import { IStoreState } from "shared/types";
 import RollbackButtonContainer from "../../../containers/RollbackButtonContainer";
 import { hapi } from "../../../shared/hapi/release";
 import itBehavesLike from "../../../shared/specs";
 import * as url from "../../../shared/url";
 import ConfirmDialog from "../../ConfirmDialog";
-
-import { IRelease } from "shared/types";
 import AppControls, { IAppControlsProps } from "./AppControls";
 import UpgradeButton from "./UpgradeButton";
 
+const mockStore = configureMockStore([thunk]);
```

Comment:
I am a bit confused, the `AppControls` component is still a class, why all these changes are needed? Is it because of the change in `react-redux`?

Reply:
ah, you commented it in the PR:

> it seems that even our class-based components are no longer classes after wrapping - at least, I had to remove calls to wrapper.state()

that's weird...